### PR TITLE
CXX-1348 Force enable deprecation attributes

### PR DESCRIPTION
**Pull request by acmorrow**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,6 +40,16 @@ else()
   message(WARNING "Unknown compiler... recklessly proceeding without a version check")
 endif()
 
+# All of our target compilers support the deprecated
+# attribute. Normally, we would just let the GenerateExportHeader
+# subsystem do this via configure check, but there appears to be a
+# CMake bug where if -Werror is set on the command line, it breaks the
+# configure check, and we end up not configuring the macro. That means
+# that we end up not being able to turn deprecation warnings into
+# errors. Instead, since we know all oru platforms offer the feature,
```
Comment (derickr) [inline CMakeLists.txt]: "oru" => "our"

Comment (acmorrow) [reply CMakeLists.txt]: Fixed.

**Comment by acmorrow** [comment]
@derickr - Per other comments, not holding this for travis.

**Comment by hanumantmk** [comment]
lgtm

**Pull request merged**